parlai/chat_service/services/terminal_chat/client.py

```diff
@@ -33,43 +29,83 @@ def prBlueBG(text):
 
 
 def on_message(ws, message):
+    """
+    Prints the incoming message from the server.
+
+    :param ws: a WebSocketApp
+    :param message: json with 'text' field to be printed
+    """
     incoming_message = json.loads(message)
     print("\033[0m\nBot: " + incoming_message['text'], "\033[44m\n")
 
 
 def on_error(ws, error):
+    """
+    Prints an error, if occurs.
+
+    :param ws: WebSocketApp
+    :param error: An error
+    """
     print(error)
 
 
 def on_close(ws):
+    """
+    Cleanup before closing connection.
+
+    :param ws: WebSocketApp
+    """
     # Reset color formatting if necessary
     print("\033[0m")
     print("Connection closed")
 
 
+def _run(ws, id):
+    """
+    Takes user input and sends it to a websocket.
+
+    :param ws: websocket.WebSocketApp
+    """
+    while True:
+        x = input("\033[44m Me: ")
+        print("\033[0m", end="")
+        data = {}
+        data['id'] = id
+        data['text'] = x
+        json_data = json.dumps(data)
+        ws.send(json_data)
+        time.sleep(1)
+        if x == "[DONE]":
+            break
+    ws.close()
+
+
 def on_open(ws):
-    id = get_rand_id()
+    """
+    Starts a new thread that loops, taking user input and sending it to the websocket.
+
+    :param ws: websocket.WebSocketApp that sends messages to a terminal_manager
+    """
+    id = _get_rand_id()
+    threading.Thread(target=_run, args=(ws, id)).start()
 
-    def run(*args):
-        while True:
-            x = input("\033[44m Me: ")
-            print("\033[0m", end="")
-            data = {}
-            data['id'] = id
-            data['text'] = x
-            json_data = json.dumps(data)
-            ws.send(json_data)
-            time.sleep(1.5)
-            if x == "[DONE]":
-                break
-        ws.close()
 
-    thread.start_new_thread(run, ())
+def setup_args():
+    """
+    Set up args, specifically for the port number.
+
+    :return: A parser that parses the port from commandline arguments.
+    """
+    parser = ParlaiParser(False, False)
+    parser.add_terminal_chat_args(is_client=True)
+    return parser.parse_args()
 
 
 if __name__ == "__main__":
+    opt = setup_args()
+    port = opt.get('port', 34596)
     ws = websocket.WebSocketApp(
-        "ws://localhost:35496/websocket",
+        "ws://localhost:{}/websocket".format(port),
```

Comment:
just a clarification is this the same port that the server uses?

Reply:
Yeah it should be. It's whatever port you pass into `--port`, and you'll get an error if there's no server running on the port you pass.